crates/api/src/comment.rs

```diff
@@ -96,23 +86,20 @@ impl Perform for SaveComment {
     };
 
     if data.save {
-      let save_comment = move |conn: &'_ _| CommentSaved::save(conn, &comment_saved_form);
-      if blocking(context.pool(), save_comment).await?.is_err() {
+      let save_comment = CommentSaved::save(&&context.pool.get().await?, &comment_saved_form);
+      if save_comment.is_err() {
         return Err(ApiError::err("couldnt_save_comment").into());
       }
     } else {
-      let unsave_comment = move |conn: &'_ _| CommentSaved::unsave(conn, &comment_saved_form);
-      if blocking(context.pool(), unsave_comment).await?.is_err() {
+      let unsave_comment = CommentSaved::unsave(&&context.pool.get().await?, &comment_saved_form);
+      if unsave_comment.is_err() {
         return Err(ApiError::err("couldnt_save_comment").into());
       }
     }
 
     let comment_id = data.comment_id;
     let person_id = local_user_view.person.id;
-    let comment_view = blocking(context.pool(), move |conn| {
-      CommentView::read(conn, comment_id, Some(person_id))
-    })
-    .await??;
+    let comment_view = CommentView::read(&&context.pool.get().await?, comment_id, Some(person_id))?;
```

Comment:
You can probably simplify this further by adding a method on context which directly returns a &PgConnection or whatever, so you can just write `context.conn().await?`.